cmd/assaf-http/main.go

```diff
@@ -252,54 +252,66 @@ func subscribe(queueName string, sessions chan chan models.Session, tracer opent
 
 		log.Printf("subscribed...")
 
-		defaultHeaders := make(map[string]interface{})
-		defaultHeaders["uber-trace-id"] = "stam"
 		for n_msg := range deliveries {
-			go returnResponse(n_msg, requests, defaultHeaders, tracer)
+			go returnResponse(n_msg, requests, tracer)
 		}
 	}
 }
-
-func returnResponse(c_msg amqp.Delivery, requests *Reqs, defaultHeaders map[string]interface{}, tracer opentracing.Tracer) error {
+func returnResponse(c_msg amqp.Delivery, requests *Reqs, tracer opentracing.Tracer) error {
 	routingKey, cerr := strconv.Atoi(c_msg.RoutingKey)
 	if cerr != nil {
 		log.Println("routing key was nil", cerr)
 	}
 
 	consumeRequest, ok := requests.Load(routingKey)
-
 	if ok == false {
 		return errors.New("request not found")
 	}
+
+	sp, e := getOrCreateSpan(c_msg, tracer)
+	defer sp.Finish()
+	if e != nil {
+		log.Printf("Span was not created %v", e)
+	}
+	c_msg.Ack(false)
+	log.Printf("got request id: %d", consumeRequest.ReqId)
+
+	_http_status := c_msg.Headers["http-status"]
+	http_status := int32(200)
+	if _http_status != nil {
+		http_status = _http_status.(int32)
+	}
+
+	select {
+	case <-consumeRequest.ReqChannel:
+		// This is done in order to no send on a closed channel which leads to panics
+		return errors.New("request closed before response")
+	case consumeRequest.ReqChannel <- models.Message{Body: c_msg.Body, RoutingKey: c_msg.RoutingKey, SpanCtx: sp.Context(), HttpStatus: http_status}:
```

Comment:
we are passing more and more data with the request, we might move to a map instead of individual data pieces